### ISSUE TITLE
trailing comma in reports lead to invalid json

### ISSUE THREAD

**Issue by surli** [revapi/revapi#159]
Hi,

revapi currently propose by default this kind of JSON snippet in case of failing check:
```
{
  "code": "java.class.nonFinalClassInheritsFromNewClass",
  "old": "class org.xwiki.rendering.macro.footnote.FootnoteMacroParameters",
  "new": "class org.xwiki.rendering.macro.footnote.FootnoteMacroParameters",
  "superClass": "org.xwiki.rendering.macro.footnote.ParameterNode",
  "justification": <<<<< ADD YOUR EXPLANATION FOR THE NECESSITY OF THIS CHANGE >>>>>
},
```
as you can see this snippet shows a trailing comma, that shouldn't be kept in the final JSON.
You can see in https://github.com/xwiki/xwiki-rendering/commit/86f9ab36f7519ccffeacf131f95d1907b762857d that we had to remove the trailing comma, because we use this JSON to generate some part of our release notes and it was failing. 
So it seems that revapi somehow fixes the wrong JSON automatically: would that be possible to make this fix optional, so we would have a failing build in case of an invalid JSON?

**Comment by metlos** [comment]
Interesting. Revapi uses the parser of https://github.com/jbossas/jboss-dmr which probably ignores these kinds of errors. Would you be able to confirm that `ModelNode.fromJSONString(String)` of JBoss DMR exhibits this, too?

**Comment by surli** [comment]
> Would you be able to confirm that ModelNode.fromJSONString(String) of JBoss DMR exhibits this, too?

I tested the following really simple piece of code:
```
public static String jsonString = "{\n"
        + "  \"code\": \"java.class.nonFinalClassInheritsFromNewClass\",\n"
        + "  \"old\": \"class org.xwiki.rendering.macro.footnote.FootnoteMacroParameters\",\n"
        + "  \"new\": \"class org.xwiki.rendering.macro.footnote.FootnoteMacroParameters\",\n"
        + "  \"superClass\": \"org.xwiki.rendering.macro.footnote.ParameterNode\",\n"
        + "  \"justification\": \"foobar\"\n"
        + "},\n";

    public static void main(String[] args)
    {
        ModelNode modelNode = ModelNode.fromJSONString(jsonString);
        System.out.println(modelNode.toJSONString(true));
    }
```

I obtain the following error:
> Exception in thread "main" java.lang.IllegalArgumentException: Unexpected content following the DMR stream

No error if I remove the trailing comma.

**Comment by metlos** [comment]
Ok, maybe a simple upgrade will do then. Revapi currently uses the old jboss-dmr 1.2.0.Final.

**Comment by surli** [comment]
I made another test with the following string:
```
public static String jsonString = "{\n"
        + "              \"revapi\": {\n"
        + "                \"filter\": {\n"
        + "                  \"archives\": {\n"
        + "                      \"exclude\": [\"org\\\\.xwiki\\\\.commons:.*:.*\"]\n"
        + "                    }\n"
        + "                },\n"
        + "                \"ignore\" : [\n"
        + "                  {\n"
        + "                    \"code\": \"java.method.parameterTypeParameterChanged\",\n"
        + "                    \"old\": \"parameter void org.xwiki.rendering.wikimodel.xhtml.impl.IgnoreElementRule::<init>(===java.util.function.Predicate<org.xwiki.rendering.wikimodel.xhtml.impl.TagContext>===, boolean)\",\n"
        + "                    \"new\": \"parameter void org.xwiki.rendering.wikimodel.xhtml.impl.IgnoreElementRule::<init>(===java.util.function.Predicate<org.xwiki.rendering.wikimodel.xhtml.impl.IgnoreElementRule>===, boolean)\",\n"
        + "                    \"justification\": \"Change the predicate to use directly the current object. The IgnoreElementRule class is still an unstable API\"\n"
        + "                  },\n"
        + "                  {\n"
        + "                    \"code\": \"java.method.numberOfParametersChanged\",\n"
        + "                    \"old\": \"method void org.xwiki.rendering.wikimodel.xhtml.impl.IgnoreElementRule::switchRule(org.xwiki.rendering.wikimodel.xhtml.impl.TagContext)\",\n"
        + "                    \"new\": \"method void org.xwiki.rendering.wikimodel.xhtml.impl.IgnoreElementRule::switchRule(org.xwiki.rendering.wikimodel.xhtml.impl.TagContext, boolean)\",\n"
        + "                    \"justification\": \"Call the switch method with the info if it is from a begin or an end element. The IgnoreElementRule class is still an Unstable API.\"\n"
        + "                  },\n"
        + "                ]\n"
        + "              }}";
```

This time I obtain:
> Exception in thread "main" java.lang.IllegalArgumentException: Expecting OBJECT_START or LIST_START or PROPERTY_START or STRING or EXPRESSION or BYTES or NUMBER or BOOLEAN or TYPE or UNDEFINED

Same as above it doesn't reproduce if I fix the JSON.

**Comment by surli** [comment]
> Ok, maybe a simple upgrade will do then. Revapi currently uses the old jboss-dmr 1.2.0.Final.

Indeed, with 1.2.0.Final I don't have any issue.

**Comment by metlos** [comment]
fixed in `revapi-0.11.2` used in `revapi-maven-plugin-0.11.2` and `revapi-java-0.20.0`.
